scalafix-rules/src/main/scala/scalafix/internal/rule/RedundantTokens.scala

```diff
@@ -0,0 +1,33 @@
+package scalafix.internal.rule
+
+import scala.meta._
+
+import metaconfig.Configured
+import scalafix.util.TokenList
+import scalafix.v1._
+
+class RedundantTokens(config: RedundantTokensConfig)
+    extends SemanticRule("RedundantTokens") {
```

Comment:
did you make it semantic for further iterations? I think the name of the rule itself calls for a syntactic one
```suggestion
    extends SyntacticRule("RedundantTokens") {
```

Reply:
Yeah, I was thinking of things like removal of `override` for implementation of abstract members (though I wouldn't want to enable that by default) - should we have two separate rules, a semantic one and a syntactic one?

Reply:
Having 2 rules sounds a bit extreme. On the other hand, starting with a syntactic one and upgrading to a semantic later would be a breaking change for clients that do not use a build tool integration. I would say this is still acceptable.

I can think of a few other useful **syntactic** rewrites (removing unnecessary string interpolators, empty bodies, etc). The question then is whether such rewrites wouldn't be better (only) in scalafmt. I do see potential in syntactic rules, but mostly for community ones actually.

Any opinion @olafurpg @mlachkar ?

Reply:
I would lean towards making most functionality available through syntactic rules as possible. I don’t think it’s extreme to split a rule between its syntactic parts and semantic. Worst case, users add two rules to their config file. However, this makes sense as long as the syntactic rules are useful on their own.

Reply:
I think we should start with a syntactic rule, and we can decide later if you change it to a semantic one, or create another new one. I do like to group rules when they offer the same functionality: for example Remove Unused can remove 4 kind of warnings, and it quite possible to built in new capacities without requiring from the user to add new rules.

Reply:
Shall we call this one `RedundantSyntax` rather than `RedundantTokens` then?

Reply:
Good idea!

Reply:
:+1: